src/main/cpp/RLBotInterface/BallPrediction/inc/vec.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <math.h>
+#include <iostream>
+#include <initializer_list>
+
+template < int n > 
+class vec {
+
+  float data[n];
+
+  public:
+
+    vec(){} 
+
+    vec(float value){
+      for (int i = 0; i < n; i++) {
+        data[i] = value;
+      }
+    } 
+
+    vec(std::initializer_list< float > args){
+      int i = 0;
+      for (float arg : args) {
+        data[i++] = arg;
+      }
+    }
+
+    float & operator[](const int i){
+      return data[i];
+    }
+
+    float operator[](const int i) const{
+      return data[i];
+    }
+
+    float & operator()(const int i){
+      return data[i];
+    }
+
+    float operator()(const int i) const{
+      return data[i];
+    }
+
+    // elementwise addition
+    vec < n > operator+(const vec < n > & other) const{
+      vec < n > v;
+      for(int i = 0; i < n; i++){
+        v[i] = data[i] + other[i];
+      }
+      return v;
+    }
+
+    // elementwise multiplication
+    vec < n > operator*(const vec < n > & other) const{
+      vec < n > v;
+      for(int i = 0; i < n; i++){
+        v[i] = data[i] * other[i];
+      }
+      return v;
+    }
+
+	// elementwise multiplication
+	vec < n > operator*(const float other) const {
+		vec < n > v;
+		for (int i = 0; i < n; i++) {
+			v[i] = data[i] * other;
+		}
+		return v;
+	}
+
+
+    // in-place elementwise addition
+    void operator+=(const vec < n > & other){
+      for(int i = 0; i < n; i++){
+        data[i] += other[i];
+      }
+    }
+
+    // unary minus
+    vec < n > operator-() const{
+      vec < n > v;
+      for(int i = 0; i < n; i++){
+        v[i] = -data[i];
+      }
+      return v;
+    }
+
+    // elementwise subtraction
+    vec < n > operator-(const vec < n > & other) const{
+      vec < n > v;
+      for(int i = 0; i < n; i++){
+        v[i] = data[i] - other[i];
+      }
+      return v;
+    }
+
+    // in-place elementwise subtraction
+    void operator-=(const vec < n > & other){
+      for(int i = 0; i < n; i++){
+        data[i] -= other[i];
+      }
+    }
+
+    // in-place scalar addition
+    void operator+=(const float other){
+      for(int i = 0; i < n; i++){
+        data[i] += other;
+      }
+    }
+
+    // in-place scalar multiplication
+    void operator*=(const float other){
+      for(int i = 0; i < n; i++){
+        data[i] *= other;
+      }
+    }
+
+    // in-place scalar division
```

Comment:
good comments. :)
Can we make them upper case and end with a period like the other comments